### ISSUE TITLE
to_json fails on inventory

### ISSUE THREAD

**Issue by dch** [ansible/ansible#10394]
``` yaml
# inventory all the things

---
- name: run inventory
  setup:

- name: fails with error
  local_action: debug msg="{{ hostvars[inventory_hostname] | to_json}}"
```

`A variable inserted a new parameter into the module args. Be sure to quote variables if they contain equal signs (for example: "{{var}}").`

there are a large number of other to_json issues noted, but AFAICT this is unique.

NB I'm away until mid April so no response.

**Comment by dch** [comment]
OSX python 2.7.6, tested with 1.8.4 and HEAD.

**Comment by dch** [comment]
fails on FreeBSD 10.1R amd64 with:

```
    ansible: 1.8.4 [FreeBSD]
    py27-pycrypto: 2.6.1_1 [FreeBSD]
    gmp: 5.1.3_2 [FreeBSD]
    py27-setuptools27: 5.5.1_1 [FreeBSD]
    py27-paramiko: 1.14.0 [FreeBSD]
    py27-ecdsa: 0.11_1 [FreeBSD]
    py27-yaml: 3.11 [FreeBSD]
    py27-Jinja2: 2.7.3 [FreeBSD]
    py27-MarkupSafe: 0.23 [FreeBSD]
    py27-Babel: 1.3_2 [FreeBSD]
    py27-pytz: 2014.10,1 [FreeBSD]
```

**Comment by bcoca** [comment]
I'm not clear what the problem is, can you rephrase the bug?

**Comment by dch** [comment]
sorry for the delay, holidays intervened :-).

 `hostvars[inventory_hostname] | to_yaml` works, whereas  `hostvars[inventory_hostname] | to_json`  or `to_nice_json` doesn't. I'd expect that any valid yaml output should also be possible in json, using python's as a 2nd pass on the produced yaml does work, however.

e.g.:

``` yml
# inventory all the things
---
- hosts: all
  tasks:
    - setup:
    - local_action:
        copy
        content="{{ hostvars[inventory_hostname] | to_yaml}}"
        dest=config/{{ inventory_hostname }}.yaml
    - local_action:
        copy
        content="{{ hostvars[inventory_hostname] | to_json}}"
        dest=config/{{ inventory_hostname }}.json
```

produces the following:

```
PLAY [all] ********************************************************************

GATHERING FACTS ***************************************************************
ok: [localhost]

TASK: [setup ] ****************************************************************
ok: [localhost]

TASK: [copy content="{{ hostvars[inventory_hostname] | to_yaml}}" dest=config/{{ inventory_hostname }}.yaml] ***
changed: [localhost -> 127.0.0.1]

TASK: [copy content="{{ hostvars[inventory_hostname] | to_json}}" dest=config/{{ inventory_hostname }}.json] ***
fatal: [localhost -> 127.0.0.1] => A variable inserted a new parameter into the module args. Be sure to quote variables if they contain equal signs (for example: "{{var}}").

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
           to retry, use: --limit @/home/ansible/inv.retry

localhost                  : ok=3    changed=1    unreachable=1    failed=0
```

Converting yaml to json & verifying with yajl 2 parser comes out fine:

```
python -c 'import sys, yaml, json; json.dump(yaml.load(sys.stdin), sys.stdout, indent=4)' < config/localhost.yaml | json_verify
  JSON is valid
```

I'm trying to push the resulting config directly into CouchDB (a JSON document store) using the uri module, and this is breaking all the things. Now using FreeBSD 10.1-p9 amd64 & ansible 1.9.0:

```
% pkg info -q |egrep -i 'py|ansib'
ansible-1.9.0.1
py27-Babel-1.3_2
py27-Jinja2-2.7.3
py27-MarkupSafe-0.23
py27-asn1-0.1.7,1
py27-asn1-modules-0.0.5_1
py27-cairo-1.10.0_2
py27-carbon-0.9.12_1
py27-cffi-0.8.6_3
py27-characteristic-14.1.0
py27-cryptography-0.7.2
py27-django-1.7.7
py27-django-tagging-0.3.2
py27-docutils-0.12
py27-ecdsa-0.11_1
py27-graphite-web-0.9.12_2
py27-openssl-0.14
py27-paramiko-1.14.0
py27-pycparser-2.10
py27-pycrypto-2.6.1_1
py27-pygments-2.0.2
py27-pytz-2014.10,1
py27-service_identity-14.0.0
py27-setuptools27-5.5.1_1
py27-six-1.8.0
py27-sphinx-1.2.3
py27-sqlite3-2.7.9_6
py27-thrift-0.9.1,1
py27-twistedCore-15.0.0_1
py27-txamqp-0.3_3
py27-whisper-0.9.12
py27-yaml-3.11
python-2.7_2,2
python2-2_3
python27-2.7.9_1
```

**Comment by jimi-c** [comment]
Hi, this situation is no longer present in the devel branch (which will be ansible 2.0). Here is the output using a version of your example above:

<pre>
# cat 10394.yml
- hosts: localhost
  tasks:
    - local_action:
        copy
        content="{{ hostvars[inventory_hostname] | to_yaml}}"
        dest=/tmp/10394_{{ inventory_hostname }}.yaml
    - local_action:
        copy
        content="{{ hostvars[inventory_hostname] | to_json}}"
        dest=/tmp/10394_{{ inventory_hostname }}.json
# ansible-playbook -vv 10394.yml 
(output truncated due to huge amount of text...)
PLAY RECAP **********************************************************************
localhost                  : ok=3    changed=2    unreachable=0    failed=0   
# ll /tmp/10394_localhost.*
-rw-r--r--. 1 root root 48360 Jul  4 23:33 /tmp/10394_localhost.json
-rw-r--r--. 1 root root 45147 Jul  4 23:33 /tmp/10394_localhost.yaml
</pre>

If you continue seeing any problems related to this issue, or if you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!
